File: bnim2219-spring/src/main/java/edu/bbte/idde/bnim2219/spring/model/Subtask.java
package edu.bbte.idde.bnim2219.spring.model;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
@ToString(callSuper = true)
@Entity
public class Subtask extends BaseEntity {
    @Column(nullable = false)
    private String name;

    @ManyToOne
    @JoinColumn(name = "chore_id", nullable = false)
    private Chore chore;
}

File: src/model/subtask.py
from sqlalchemy.orm import relationship

from src.model.base_entity import BaseEntity
from src.db import db


class Subtask(BaseEntity):
    name = db.Column(db.String(255), nullable=False)
    choreId = db.Column(db.Integer, db.ForeignKey('chore.id'), nullable=False)
    chore = relationship('Chore', back_populates='subtasks')
